spring-core/src/main/java/org/springframework/core/MethodParameter.java

```diff
@@ -759,7 +765,47 @@ else if (ctor != null) {
 			}
 			return false;
 		}
-		
-	}
 
+		/**
+		 * Returns a return type of a method using Kotlin Reflection API.
+		 * Introduced to support suspending functions.
+		 */
+		@NonNull
+		static private Class<?> getReturnType(Method method) {
+		    if (isSuspend(method)) {
+		        return getSuspendReturnType(method).resolve();
```

Comment:
`resolve()` can return `null` but `Class<?> getReturnType(Method method)` is expected to return non null values. Maybe we should use `resolve(Class<?> fallback)` variant or assert the returned class is not null before returning it if we are sure that will be the case here?

Reply:
Assertion added. This should never be `null`.